pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -453,20 +448,34 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 		if len(virtualHostWrapper.Routes) == 0 {
 			continue
 		}
-		virtualHosts := make([]*route.VirtualHost, 0, len(virtualHostWrapper.VirtualServiceHosts)+len(virtualHostWrapper.Services))
 
+		virtualHost := &route.VirtualHost{
+			Routes:                     virtualHostWrapper.Routes,
+			IncludeRequestAttemptCount: true,
+		}
+		var domains []string
 		for _, hostname := range virtualHostWrapper.VirtualServiceHosts {
-			if vhost := buildVirtualHost(hostname, virtualHostWrapper, nil); vhost != nil {
-				virtualHosts = append(virtualHosts, vhost)
+			if virtualHost.Name == "" {
+				virtualHost.Name = util.DomainName(hostname, virtualHostWrapper.Port)
+			}
+			if dm := buildVirtualHostDomains(hostname, virtualHostWrapper, nil); len(dm) != 0 {
+				domains = append(domains, dm...)
```

Comment:
I think here we need to dedup domains

Reply:
need to dedup in case there are two hosts? :
`test.ns.svc.cluster.local`
and `test.ns`

Reply:
They are different when matching routes. For example, like this envoy virtualhost:
```
{
    "name": "test.ns.svc.cluster.local:9080",
    "domains": [
        "test.ns.svc.cluster.local",
        "test.ns",
    ],
}
```